### PR TITLE
Add ability to ignore certain errors

### LINKED ISSUE #927: ERR_CONNECTION_TIMED_OUT
**macOS version:**    Catalina (10.15)
**Kap version:**      3.3.1

#### Workaround

Unhandled Promise Rejection
Error: net::ERR_CONNECTION_TIMED_OUT
    at SimpleURLLoaderWrapper.<anonymous> (electron/js2c/browser_init.js:2519:21)

### PR DESCRIPTION
**Pull request by karaggeorge**
Fixes #927 

Added ability to ignore certain errors based on regex. Also added regex for the one in the issue above

### CODE REVIEW

File: main/utils/errors.js [hunk 1]
```diff
@@ -14,6 +14,12 @@ const {showDialog} = require('../dialog');
 
 const MAX_RETRIES = 10;
 
+const ERRORS_TO_IGNORE = [
+  /net::ERR_CONNECTION_TIMED_OUT/m
```
Comment (sindresorhus) [inline main/utils/errors.js:18]: Why the `m` flag?

Comment (karaggeorge) [reply main/utils/errors.js]: Oops, good catch

**Comment by sindresorhus** [comment]
Some more errors we should ignore:
- https://sentry.io/organizations/wulkano-l0/issues/1346464044/?project=255536&query=is%3Aunresolved
- https://sentry.io/organizations/wulkano-l0/issues/845277246/?project=255536&query=is%3Aunresolved

**Comment by sindresorhus** [comment]
We should also `try-catch` the auto-updater call and ignore *any* error from it.

**Pull request merged**